### PR TITLE
Problem: tx hash may be incorrect in event parsing

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- Due to bug https://github.com/evmos/ethermint/issues/1175, the first `ethereum_tx` event emitted in ante handler may be incorrect,
  Althrough the root casue is fixed, but to avoid resync we can do a workaround in event parsing to override with the second one.

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 - [#526](https://github.com/crypto-org-chain/cronos/pull/526) Fix tendermint duplicated tx issue.
 - [#584](https://github.com/crypto-org-chain/cronos/pull/584) Validate eth tx hash in ante handler and fix tx hashes returned in some JSON-RPC apis.
 - [#587](https://github.com/crypto-org-chain/cronos/pull/587) Unlucky tx patch cmd recompute eth tx hash.
+- []() Workaround the tx hash issue in event parsing.
```
Comment (yihuang) [inline CHANGELOG.md:12]:
```suggestion
- [#595](https://github.com/crypto-org-chain/cronos/pull/595) Workaround the tx hash issue in event parsing.
```

File: go.mod [hunk 2]
```diff
@@ -164,7 +164,7 @@ replace (
 
 	github.com/tendermint/tendermint => github.com/tendermint/tendermint v0.34.20-rc0.0.20220630153554-17f4ea3680de
 
-	github.com/tharsis/ethermint => github.com/crypto-org-chain/ethermint v0.10.0-cronos.0.20220715022408-e190fec288d2
+	github.com/tharsis/ethermint => github.com/crypto-org-chain/ethermint v0.10.0-cronos.0.20220719024136-4658529abbf5
```
Comment (yihuang) [inline go.mod:167]: https://github.com/crypto-org-chain/ethermint/compare/e190fec288d2...4658529abbf5

**Comment by yihuang** [comment]
Verified in production.

**Comment by adu-crypto** [comment]
we don't validate other fields like `Hash`/`Size_` in `MsgEthereumTx`.
is this intended to avoid consensus problems?

**Comment by yihuang** [comment]
> we don't validate other fields like `Hash`/`Size_` in `MsgEthereumTx`. is this intended to avoid consensus problems?

yeah, it's a non-breaking backport to v0.7, in main branch, it's in the `ValidateBasic`.

**Comment by adu-crypto** [comment]
I mean in `vbd.Antehandle` only `From` is checked:
```go
	for _, msg := range protoTx.GetMsgs() {
		msgEthTx, ok := msg.(*evmtypes.MsgEthereumTx)
		if !ok {
			return ctx, sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "invalid message type %T, expected %T", msg, (*evmtypes.MsgEthereumTx)(nil))
		}

		// Validate `From` field
		if msgEthTx.From != "" {
			return ctx, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "invalid From %s, expect empty string", msgEthTx.From)
		}

		txGasLimit += msgEthTx.GetGas()
...
```
https://github.com/evmos/ethermint/blob/main/app/ante/eth.go#L436-L438

**Comment by yihuang** [comment]
> I mean in `vbd.Antehandle` only `From` is checked:
> 
> ```go
> 	for _, msg := range protoTx.GetMsgs() {
> 		msgEthTx, ok := msg.(*evmtypes.MsgEthereumTx)
> 		if !ok {
> 			return ctx, sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "invalid message type %T, expected %T", msg, (*evmtypes.MsgEthereumTx)(nil))
> 		}
> 
> 		// Validate `From` field
> 		if msgEthTx.From != "" {
> 			return ctx, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "invalid From %s, expect empty string", msgEthTx.From)
> 		}
> 
> 		txGasLimit += msgEthTx.GetGas()
> ...
> ```
> 
> https://github.com/evmos/ethermint/blob/main/app/ante/eth.go#L436-L438

`Hash` and `Size_` is checked in https://github.com/evmos/ethermint/blob/main/x/evm/types/msg.go#L171

**Review by adu-crypto — APPROVED** [review]

**Pull request merged**